var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1011,8 +1011,7 @@ def setup_dependent_run_environment(self, env, dependent_spec):
         """
         for d in dependent_spec.traverse(deptype=('run'), root=True):
             if d.package.extends(self.spec):
-                env.prepend_path('PYTHONPATH', join_path(
-                    d.prefix, self.site_packages_dir))
+                env.prepend_path('PYTHONPATH', d.package.site_packages_dir)
```

Comment:
You'll also want to modify `setup_dependent_build_environment` above.

Reply:
Unfortunately we are not ensured that every method has a valid `site_packages_dir` directory pre-install. `clingo` is one example where we need to inspect the installation prefix to get the location where the Python extension is installed.

Reply:
Outdated